src/native/minipal/time.c

```diff
@@ -77,6 +86,55 @@ int64_t minipal_hires_ticks(void)
 #endif
 }
 
+int64_t minipal_lowres_ticks(void)
+{
+#if HAVE_CLOCK_GETTIME_NSEC_NP
+    return  (int64_t)clock_gettime_nsec_np(CLOCK_UPTIME_RAW) / (int64_t)(tccMilliSecondsToNanoSeconds);
+#elif HAVE_CLOCK_MONOTONIC
+    struct timespec ts;
+
+    // emscripten exposes CLOCK_MONOTONIC_COARSE but doesn't implement it
+#if HAVE_CLOCK_MONOTONIC_COARSE && !defined(__EMSCRIPTEN__)
+    // CLOCK_MONOTONIC_COARSE has enough precision for GetTickCount but
+    // doesn't have the same overhead as CLOCK_MONOTONIC. This allows
+    // overall higher throughput. See dotnet/coreclr#2257 for more details.
+
+    const clockid_t clockType = CLOCK_MONOTONIC_COARSE;
+#else
+    const clockid_t clockType = CLOCK_MONOTONIC;
+#endif
+
+    int result = clock_gettime(clockType, &ts);
+    if (result != 0)
+    {
+#if HAVE_CLOCK_MONOTONIC_COARSE && !defined(__EMSCRIPTEN__)
+        assert(!"clock_gettime(CLOCK_MONOTONIC_COARSE) failed");
+#else
+        assert(!"clock_gettime(CLOCK_MONOTONIC) failed");
+#endif
+    }
+
+    return ((int64_t)(ts.tv_sec) * (int64_t)(tccSecondsToMilliSeconds)) + ((int64_t)(ts.tv_nsec) / (int64_t)(tccMilliSecondsToNanoSeconds));
+#elif HAVE_GETHRTIME
+    return (int64_t)(gethrtime () / tccMilliSecondsToNanoSeconds);
+#elif HAVE_READ_REAL_TIME
+    timebasestruct_t tb;
+    read_real_time (&tb, TIMEBASE_SZ);
+    if (time_base_to_time (&tb, TIMEBASE_SZ) != 0)
+    {
+        assert(!"time_base_to_time() failed");
+    }
+    return (tb.tb_high * tccSecondsToMilliSeconds) + (tb.tb_low / tccMilliSecondsToNanoSeconds);
+#else
+    struct timeval tv;
+    if (gettimeofday(&tv, NULL) != 0)
+    {
+        assert(!"gettimeofday() failed\n");
+    }
+    return ((int64_t)(tv.tv_sec) * (int64_t)(tccSecondsToMilliSeconds)) + ((int64_t)(tv.tv_usec) / (int64_t)(tccMilliSecondsToMicroSeconds));
+#endif
```

Comment:
Are these fallback paths still needed now? They are copied from mono. Since `minipal_hires_ticks` is requiring either `clock_gettime_nsec_np` or `CLOCK_MONOTONIC`, other paths will never be compiled successfully.
Searching for the symbols results to Oracle and IBM wiki, I assume they are used for some old UNIX? `gettimeofday` is from older POSIX and considered deprecated, we should probably never target a UNIX without `clock_gettime`.